pkg/pipelines/config/config.go

```diff
@@ -116,21 +112,29 @@ type Environment struct {
 	Apps      []*Application `json:"apps,omitempty"`
 	// TODO: this should check that there is 0 or 1 CICD environment in the
 	// manfifest.
-	IsCICD   bool `json:"cicd,omitempty"`
-	IsArgoCD bool `json:"argo,omitempty"`
+}
+
+//Config are the Config environments that constitute the argocd and cicd environment
+type Config struct {
+	CICD *Cicd `json:"CICD,omitempty"`
+	Argo *Argo `json:"Argo,omitempty"`
+}
+
+//Cicd checks for the cicd environments
+type Cicd struct {
+	Namespace string `json:"namespace,omitempty"`
+}
+
+//Argo checks for the cicd environments
+type Argo struct {
+	Namespace string `json:"namespace,omitempty"`
```

Comment:
We should call it "Name" like any entities in the manifest.   We just create namespaces for them but they are names of something (like Env, App).  In this case, it is a Config.